File: server/api/src/view_models/portfolio_view_model.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

use super::{
    assets::base_models::asset::IdentifiableAssetViewModel,
    portfolio_entry_view_model::PortfolioEntryViewModel,
};

#[derive(Clone, Debug, Serialize, Deserialize, ToSchema)]
pub struct PortfolioViewModel {
    pub portfolio_entries: Vec<PortfolioEntryViewModel>,
    pub reference_asset: IdentifiableAssetViewModel,
}

// impl From<PortfolioDto> for PortfolioViewModel {
//     fn from(p: PortfolioDto) -> Self {
//         Self {
//             portfolio_entries: p.rows.into_iter().map(|x| x.into()).collect(),
//             reference_asset: p.reference_asset.into(),
//         }
//     }
// }

File: web/src/models/portfolio_view_model.ts
import { AssetViewModel } from "./asset_view_model";

export interface PortfolioAccountViewModel {
  id: string;
  name: string;
}

export interface PortfolioEntryViewModel {
  asset: AssetViewModel;
  account: PortfolioAccountViewModel;
  sum: number;
}

export interface PortfolioViewModel {
  portfolio_entries: PortfolioEntryViewModel[];
}
